code/obj/vehicle.dm

```diff
@@ -241,17 +241,119 @@ ABSTRACT_TYPE(/obj/vehicle)
 	mats = 8
 	var/weeoo_in_progress = 0
 	var/icon_weeoo_state = 2
+	var/obj/item/cell/cell
+	var/open = 0 //Cell hatch
+	var/bat_warning = 10 // battery percentage at which warning plays
+	var/playing = FALSE //movement sound cooldown
+	var/bumpin = FALSE //Same as playing but for the bump animation
+
 	soundproofing = 0
 	throw_dropped_items_overboard = 1
 	var/datum/light/light
 	ability_buttons_to_initialize = list(/obj/ability_button/weeoo)
 	var/obj/item/joustingTool = null // When jousting will be reference to lance being used
 
+
+/obj/vehicle/segway/examine() //we need to update the battery indicator every time its examined
+	desc = "Now you too can look like a complete tool in space!	The battery indicator reads [floor((cell.charge / cell.maxcharge) * 100)]% remaining."
+	. = ..()
```

Comment:
The standard way to do this is to add the `The battery indicator reads [...] remaining` in get_desc.

But also, examining a segway with no cell in it causes a runtime atm.